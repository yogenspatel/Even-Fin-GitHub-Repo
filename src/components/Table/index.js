import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { SortBy } from "../../actions";

class Table extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sortBy: "userId",
      order: "asc"
    };
  }

  sortBy = e => {
    this.props.SortBy(e.target.name, this.props.tableData, e.target.value);
    this.setState({
      sortBy: e.target.name,
      order: e.target.value
    });
  };

  renderTable() {
    const { tableData } = this.props;
    return (
      <table>
        <thead>
          <tr>
            <th>
              userId
              <select
                name="userId"
                onChange={this.sortBy}
                value={this.state.sortBy === "userId" ? this.state.order : ""}
              >
                <option value="">Order</option>
                <option value="asc">Asc</option>
                <option value="dec">Dec</option>
              </select>
            </th>
            <th>
              Title
              <select
                name="title"
                onChange={this.sortBy}
                value={this.state.sortBy === "title" ? this.state.order : ""}
              >
                <option value="">Order</option>
                <option value="asc">Asc</option>
                <option value="dec">Dec</option>
              </select>
            </th>
            <th>
              Completed ?
              <select
                name="completed"
                onChange={this.sortBy}
                value={this.state.sortBy === "completed" ? this.state.order : ""}
              >
                <option value="">Order</option>
                <option value="asc">Asc</option>
                <option value="dec">Dec</option>
              </select>
            </th>
          </tr>
        </thead>
        {tableData && tableData.length && <tbody>{this.renderRows(tableData)}</tbody>}
      </table>
    );
  }

  renderRows(data) {
    return data.map(todo => this.renderRow(todo));
  }

  renderRow(item) {
    return (
      <tr key={item.id}>
        <td>{item.userId}</td>
        <td>{item.title}</td>
        <td>{item.completed ? "Completed" : "Not Completed"}</td>
      </tr>
    );
  }
  render() {
    return <div>{this.renderTable()}</div>;
  }
}

Table.propTypes = {
  SortBy: () => {},
  tableData: PropTypes.array
};

Table.defaultProps = {
  tableData: [],
  SortBy: null
};

function mapStateToProps(state) {
  let { tableData, searchData } = state;
  tableData = searchData ? searchData : tableData;
  return { tableData };
}

export default connect(
  mapStateToProps,
  { SortBy }
)(Table);
