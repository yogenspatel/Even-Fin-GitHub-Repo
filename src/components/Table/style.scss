.orange {
    td:first-child,
    td:nth-child(2),
    td:nth-child(3),
    td:nth-child(4) {
        color: #FF4500;
    }
}
.green {
    td:first-child,
    td:nth-child(2),
    td:nth-child(3),
    td:nth-child(4) {
        color: #008000;
    }
}
