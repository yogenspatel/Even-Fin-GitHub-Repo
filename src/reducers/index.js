import { createStore, combineReducers } from "redux";
import { reducer as reduxFormReducer } from "redux-form";
import searchReducer from "./search";

const rootReducer = combineReducers({
  form: reduxFormReducer, // mounted under "form"
  searchResults: searchReducer
});

function configureStore(state = {}) {
  return createStore(rootReducer, state);
}

export { configureStore };
export default rootReducer;
