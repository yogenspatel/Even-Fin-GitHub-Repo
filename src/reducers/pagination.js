/* eslint-disable no-case-declarations */
import { PAGINATED_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case PAGINATED_DATA:
      const { payload, noOfItems, pageSize, searchData } = action;
      return { ...state, paginated_data: payload, noOfItems, pageSize, searchData };
    default:
      return state;
  }
};
