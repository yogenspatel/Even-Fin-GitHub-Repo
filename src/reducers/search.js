import { PERFORM_SEARCH } from '../actions';

export default (state = null, action) => {
  switch (action.type) {
    case PERFORM_SEARCH:
      return action.payload.data.items;
    default:
      break;
  }
  return state;
};
