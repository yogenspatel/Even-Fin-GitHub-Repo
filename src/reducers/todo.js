import { GET_TODO_DATA, SET_LOADING_STATUS, SET_DATA_ERROR, SORT_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case SET_LOADING_STATUS:
      return { ...state, status: action.status };
    case SET_DATA_ERROR:
      return { ...state, status: action.error };
    case GET_TODO_DATA:
      return action.payload ? action.payload : state;
    case SORT_DATA:
      console.log('state: ', action.sort_data);
      return action.sort_data;
    default:
      return state;
  }
};
