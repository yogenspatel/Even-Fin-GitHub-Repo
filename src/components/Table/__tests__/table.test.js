/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import { Table } from "../../Table";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { configureStore } from "../../../reducers";

const data = [
  {
    id: 2,
    title: "this is test"
  },
  {
    id: 1,
    title: "delectus aut autem"
  },
  {
    id: 3,
    title: "quis ut nam facilis et officia qui"
  },
  {
    id: 4,
    title: "quo laboriosam deleniti aut qui"
  }
];
const searchData = [
  {
    id: 2,
    title: "this is test"
  },
  {
    id: 4,
    title: "quo laboriosam deleniti aut qui"
  }
];

describe("Table Component renders correctly", () => {
  it("Table renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Table />, div);
  });

  it("Table Component with Mock Data", () => {
    const store = configureStore();
    const tree = shallow(<Table store={store} userData={data} SortBy={jest.fn} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("Table Component without mock data", () => {
    const store = configureStore();
    const tree = shallow(<Table store={store} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("Table Component with search data", () => {
    const store = configureStore();
    const tree = shallow(<Table store={store} userData={searchData} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
