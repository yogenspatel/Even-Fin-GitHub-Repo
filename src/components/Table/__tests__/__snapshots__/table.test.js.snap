// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders correctly Table Component with Mock Data 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          id
          <select
            className="custom-select"
            name="id"
            onChange={[Function]}
            value="asc"
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          userId
          <select
            className="custom-select"
            name="userId"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Title
          <select
            className="custom-select"
            name="title"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Completed ?
          <select
            className="custom-select"
            name="completed"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Orange 
          &lt;--&gt;
           Black
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="2"
      >
        <td>
          2
        </td>
        <td />
        <td>
          this is test
        </td>
        <td>
          Not Completed
        </td>
        <td>
          <button
            className="btn btn-info"
            onClick={[Function]}
          >
            Orange 
            &lt;--&gt;
             Black
             
          </button>
        </td>
      </tr>
      <tr
        key="1"
      >
        <td>
          1
        </td>
        <td />
        <td>
          delectus aut autem
        </td>
        <td>
          Not Completed
        </td>
        <td>
          <button
            className="btn btn-info"
            onClick={[Function]}
          >
            Orange 
            &lt;--&gt;
             Black
             
          </button>
        </td>
      </tr>
      <tr
        key="3"
      >
        <td>
          3
        </td>
        <td />
        <td>
          quis ut nam facilis et officia qui
        </td>
        <td>
          Not Completed
        </td>
        <td>
          <button
            className="btn btn-info"
            onClick={[Function]}
          >
            Orange 
            &lt;--&gt;
             Black
             
          </button>
        </td>
      </tr>
      <tr
        key="4"
      >
        <td>
          4
        </td>
        <td />
        <td>
          quo laboriosam deleniti aut qui
        </td>
        <td>
          Not Completed
        </td>
        <td>
          <button
            className="btn btn-info"
            onClick={[Function]}
          >
            Orange 
            &lt;--&gt;
             Black
             
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table Component renders correctly Table Component with search data 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          id
          <select
            className="custom-select"
            name="id"
            onChange={[Function]}
            value="asc"
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          userId
          <select
            className="custom-select"
            name="userId"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Title
          <select
            className="custom-select"
            name="title"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Completed ?
          <select
            className="custom-select"
            name="completed"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Orange 
          &lt;--&gt;
           Black
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="2"
      >
        <td>
          2
        </td>
        <td />
        <td>
          this is test
        </td>
        <td>
          Not Completed
        </td>
        <td>
          <button
            className="btn btn-info"
            onClick={[Function]}
          >
            Orange 
            &lt;--&gt;
             Black
             
          </button>
        </td>
      </tr>
      <tr
        key="4"
      >
        <td>
          4
        </td>
        <td />
        <td>
          quo laboriosam deleniti aut qui
        </td>
        <td>
          Not Completed
        </td>
        <td>
          <button
            className="btn btn-info"
            onClick={[Function]}
          >
            Orange 
            &lt;--&gt;
             Black
             
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table Component renders correctly Table Component without mock data 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          id
          <select
            className="custom-select"
            name="id"
            onChange={[Function]}
            value="asc"
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          userId
          <select
            className="custom-select"
            name="userId"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Title
          <select
            className="custom-select"
            name="title"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Completed ?
          <select
            className="custom-select"
            name="completed"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Order
            </option>
            <option
              value="asc"
            >
              Asc
            </option>
            <option
              value="dec"
            >
              Dec
            </option>
          </select>
        </th>
        <th>
          Orange 
          &lt;--&gt;
           Black
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          No Results found
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
